### PR TITLE
Fix http proxy ENV when TLS is enabled.

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/containerd/cri/issues/1169.

Support http proxy env.

The transport is the same with https://github.com/containerd/containerd/blob/master/cmd/ctr/commands/resolver.go#L82

Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

File: pkg/server/image_pull.go [hunk 1]
```diff
@@ -296,15 +299,14 @@ func (c *criService) getResolver(ctx context.Context, ref string, cred func(stri
 		}
 
 		if registryTLSConfig, ok := c.config.Registry.TLSConfigs[u.Host]; ok {
-			tlsTransport, err := c.getRegistryTLSTransport(ctx, registryTLSConfig)
+			transport.TLSClientConfig, err = c.getTLSConfig(registryTLSConfig)
```
Comment (mikebrow) [inline pkg/server/image_pull.go]: that's better!

File: pkg/server/image_pull.go [hunk 2]
```diff
@@ -340,3 +341,19 @@ func (c *criService) getResolver(ctx context.Context, ref string, cred func(stri
 	}
 	return resolver, desc, nil
 }
+
+// newTransport returns a new HTTP transport used to pull image.
+func newTransport() *http.Transport {
+	return &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
```
Comment (mikebrow) [inline pkg/server/image_pull.go]: looks good..

**Comment by Random-Liu** [comment]
> create some test cases unit and crictl or node that test TLS, Proxy, and other off the beaten path patterns for registry pull tests

It is hard to test this. :(

> create some common code in containerd that can be used here and in ctr.

Added TODO

**Comment by Random-Liu** [comment]
Just added a TODO, apply lgtm based on https://github.com/containerd/cri/pull/1171#pullrequestreview-248761802

**Pull request merged**